### PR TITLE
Fetch the adapter from the config

### PR DESCRIPTION
**Pull request by jcoyne**
`nil` is not a valid option for `:adapter`

### CODE REVIEW

File: lib/blacklight.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def self.repository_class
     when /::/
       connection_config[:adapter].constantize
     else
-      Blacklight.const_get("#{connection_config[:adapter]}/Repository".classify)
```
Comment (cbeer) [inline lib/blacklight.rb]: Ought we just raise an exception if the adapter isn't defined?

Comment (jcoyne) [inline lib/blacklight.rb]: If you don't do this, you get the very confusing message that class "Repository" could not be found. I think a `KeyError` does a better job pointing to the problem

Comment (cbeer) [inline lib/blacklight.rb]: I meant, raise an exception ourselves that describes the problem and how to fix it (likely a misconfiguration or a Blacklight 5 -> 6 migration error?). If you think just a key error is sufficient, fine.

**Comment by jcoyne** [comment]
@cbeer updated.

**Comment by cbeer** [comment]
:+1:

**Pull request merged**